### PR TITLE
REV-009: create agent list page

### PR DESCRIPTION
**Pull request by RaniaSAloun**
Created agent list page with the following changes:

- Created agent card as a shared component in an added common folder.
- Added the cards to the page.
- Created AgentsListBanner component holding the heading and the filter bar.
- Added the new page's route to the routes file.
- Created two separate components for the search section (mobile and desktop) since the component's design is not identical.  

Slide pages:
Desktop: https://optimumpartners.invisionapp.com/console/share/2Y34DAX6RD/812252250
Mobile: https://optimumpartners.invisionapp.com/console/share/3V3512T8G6/843334662

### CODE REVIEW

File: src/assets/icons/like.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-thumbs-up"><path d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"></path></svg>
```
Comment (Mahmoudzaid2994) [inline src/assets/icons/like.svg:1]: please add an empty line at the end of each file.

File: src/components/AgentCardsContainer/agent-cards-container.styles.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export const CardsContainer = styled.div`
+  width: 100%;
+
+  ${mq.desktopExtraMax`
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: flex-start;
+    margin: 1.25rem 5.4375rem;
+    overflow: hidden;
+    max-width: 79rem;
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/agent-cards-container.styles.js:18]: Could you order these props alphabetically

File: src/components/AgentCardsContainer/index.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { CardsContainer, Container } from './agent-cards-container.styles';
+
+/**
+ * Component holding the cards of agents, it maps agents to their cards.
+ *
+ * @return {JSX.Element} Container holding agent cards.
+ */
+function AgentCardsContainer({ filteredTarget }) {
+  const [agents, setAgents] = useState([]);
+
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/index.jsx:14]: Please add documentation for any function
```suggestion

  // fetch agent data
```

File: src/components/AgentCardsContainer/index.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { getAgents } from 'services/fake-agents-service';
+import { CardsContainer, Container } from './agent-cards-container.styles';
+
+/**
+ * Component holding the cards of agents, it maps agents to their cards.
+ *
+ * @return {JSX.Element} Container holding agent cards.
+ */
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/index.jsx:11]:
```suggestion
/**
 * Component holding the cards of agents, it maps agents to their cards.
 *
 * @param {array} filteredTarget Filtered target array.
 *
 * @return {JSX.Element} Container holding agent cards.
 */
```

File: src/components/AgentCardsContainer/index.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            key={agent.id}
+          />
+        ))}
+      </CardsContainer>
+    </Container>
+  );
+}
+
+AgentCardsContainer.propTypes = {
+  filteredTarget: PropTypes.string.isRequired,
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/index.jsx:46]:
```suggestion
  filteredTarget: PropTypes.arrayOf.isRequired,
```

File: src/components/AgentsListBanner/agents-list-banner.styles.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  margin: 0.625rem 0rem 0.625rem 0.625rem;
+  width: 1rem;
+`;
+
+export const FilterIcon = styled(filter)`
+  fill: ${colors.green};
+  width: 0.75rem;
+`;
+
+export const FilterTxt = styled.p`
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/agents-list-banner.styles.js:71]: change the name to FilterText and also please change it at the components where use it

File: src/components/AgentsListBanner/agents-list-banner.styles.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  width: 100%;
+
+  ${mq.desktopExtraMax`
+  font-size: 1.125rem;
+  padding: 0.625rem;
+  width: 100%;
+`};
+`;
+
+export const SearchBtn = styled.button`
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/agents-list-banner.styles.js:124]: change the name to SearchButton and also please change it at the components where use it

File: src/components/AgentsListBanner/index.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  Heading,
+  Input,
+  SearchBtn,
+  SearchIcon,
+  Select,
+} from './agents-list-banner.styles';
+
+/**
+ * Component showing the Banner in the the agent's page.
+ *
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:21]: add documentation for filter props

File: src/components/AgentsListBanner/index.jsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+ * @return {JSX.Element} Banner holding a paragraph and a Filter section.
+ */
+function AgentsListBanner({ filter }) {
+  const [addresses, setAddresses] = useState([]);
+
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:26]: add documentation for this function and also for all functions please

File: src/components/AgentsListBanner/index.jsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    const fetchAddresses = async () => {
+      const data = await getAddresses;
+      setAddresses(data);
+    };
+    fetchAddresses();
+  });
+
+  const handleAddressFilter = (e) => {
+    filter(e.target.value);
+  };
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:37]:
```suggestion
  const handleAddressFilter = (event) => {
    filter(event.target.value);
  };
```

File: src/components/AgentsListBanner/index.jsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+  return (
+    <Container>
+      <BannerContainer>
+        <Heading>Connect with your agent on reverifi</Heading>
+
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:43]:
```suggestion

```

File: src/components/shared/AgentCard/agent-card.styles.js [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+  width: 1.125rem;
+  height: 1.125rem;
+`;
+
+export const PhoneIcon = styled(phone)``;
+
+export const MailIcon = styled(email)``;
+
+export const AddressIcon = styled(address)``;
+
```
Comment (Mahmoudzaid2994) [inline src/components/shared/AgentCard/agent-card.styles.js:184]: use this SVG's directly at the component
```suggestion

```

File: src/components/shared/AgentCard/index.jsx [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+/**
+ * Agent Card.
+ *
+ * @param {Object} props Props passed to AgentCard.
+ * @param {string} props.address Address of agent.
+ * @param {string} props.agentName Name of agent.
+ * @param {string} props.companyName Agent's company name.
+ * @param {string} props.email Contact email of agent.
+ * @param {number} props.phoneNumber Contact number of agent.
```
Comment (Mahmoudzaid2994) [inline src/components/shared/AgentCard/index.jsx:42]:
```suggestion
 * @param {Object} props               Props passed to AgentCard.
 * @param {string} props.address       Agent's address.
 * @param {string} props.agentName     Agent's name.
 * @param {string} props.companyName   Agent's company name.
 * @param {string} props.email Contact Agent's email.
 * @param {number} props.phoneNumber   Agent's contact number.
```

**Comment by Mahmoudzaid2994** [comment]
For the Images, I think we should move them to Image folder and also use kabab case for naming

File: src/components/AgentCardsContainer/AgentCard/agent-card.styles.js [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+export const PropertyIconContainer = styled.div`
+  color: ${colors.green};
+  width: 1rem;
+`;
+
+export const PropertyTxt = styled.p`
+  color: ${colors.osloGrey};
+  font-family: Montserrat;
+  font-size: 1rem;
+  font-style: normal;
```
Comment (AseelRousan-op) [inline src/components/AgentCardsContainer/AgentCard/agent-card.styles.js:138]: we can remove this style from all the components
```suggestion
```

File: src/components/AgentCardsContainer/AgentCard/agent-card.styles.js [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+  font-weight: 500;
+  line-height: 1.25rem;
+  margin: 0 0 0 0.5rem;
+  mix-blend-mode: normal;
```
Comment (AseelRousan-op) [inline src/components/AgentCardsContainer/AgentCard/agent-card.styles.js:142]: the default mix-belnd-mode is normal, I think we can also remove this 
```suggestion
```

File: src/components/AgentsListBanner/agents-list-banner.styles.js [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+  font-style: normal;
+  font-weight: 600;
+  line-height: 2.4375rem;
+  margin: 0.9375rem 0.625rem;
+`;
+
+export const FilterSection = styled.div`
+  background: white;
+  border-radius: 1.75rem;
+  border: none;
```
Comment (AseelRousan-op) [inline src/components/AgentsListBanner/agents-list-banner.styles.js:41]:
```suggestion
```

File: src/components/AgentsListBanner/agents-list-banner.styles.js [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+  margin: 1rem 0;
+  padding: 0;
+  width: 100%;
+`;
+
+export const BannerContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0;
```
Comment (AseelRousan-op) [inline src/components/AgentsListBanner/agents-list-banner.styles.js:20]:
```suggestion
```

File: src/components/AgentsListBanner/agents-list-banner.styles.js [hunk 18]
```diff
@@ -0,0 +1,211 @@
...
+import styled from 'styled-components';
+import colors from 'styles/colors';
+import mq from 'styles/media-query';
+
+export const Container = styled.div`
+  display: flex;
+  justify-content: center;
+  margin: 0;
+  margin: 1rem 0;
+  padding: 0;
```
Comment (AseelRousan-op) [inline src/components/AgentsListBanner/agents-list-banner.styles.js:12]:
```suggestion
```

**Review by AseelRousan-op — COMMENTED** [review]
Looks good few small styles changes

File: src/components/AgentCardsContainer/AgentCard/agent-card.styles.js [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+  display: flex;
+  flex-direction: column;
+  margin: 0 0 1.5625rem;
+  overflow: hidden;
+
+  ${mq.desktopExtraMax`
+    flex-direction: row;
+    height: 18.5rem;
+    margin: 0 1rem 2rem 1rem;
+    width: 37.5rem;
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/AgentCard/agent-card.styles.js:21]:
```suggestion
  ${mq.desktop`
    flex-direction: row;
```

File: src/components/AgentCardsContainer/agent-cards-container.styles.js [hunk 20]
```diff
@@ -0,0 +1,20 @@
+import styled from 'styled-components';
+import mq from 'styles/media-query';
+
+export const Container = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/agent-cards-container.styles.js:8]:
```suggestion

export const Wrapper = styled.div`
  padding: 1rem;

  ${mq.desktop`
   flex: 0 0 50%;
   max-width: 50%;
  `};
`;

```

File: src/components/AgentCardsContainer/agent-cards-container.styles.js [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+  width: 100%;
+
+  ${mq.desktopExtraMax`
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: flex-start;
+    margin: 1.25rem 5.4375rem;
+    max-width: 79rem; 
+    overflow: hidden;
+  `};
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/agent-cards-container.styles.js:19]:
```suggestion
   ${mq.desktop`
    display: flex;
    flex-wrap: wrap;
    max-width: 79rem; 
    padding: 0.5rem;
  `};
```

File: src/components/AgentCardsContainer/index.jsx [hunk 22]
```diff
@@ -0,0 +1,41 @@
+import AgentCard from 'components/AgentCardsContainer/AgentCard';
+import React, { useEffect, useState } from 'react';
+import getAgents from 'services/fake-agents-service';
+import { CardsContainer, Container } from './agent-cards-container.styles';
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/index.jsx:4]:
```suggestion
import {
  CardsContainer,
  Container,
  Wrapper,
} from './agent-cards-container.styles';```

File: src/components/AgentCardsContainer/index.jsx [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+        {agents.map((agent) => (
+          <AgentCard
+            address={agent.address}
+            agentImg={agent.agentImg}
+            agentName={agent.agentName}
+            companyName={agent.companyName}
+            email={agent.email}
+            key={agent.id}
+            phoneNumber={agent.phoneNumber}
+          />
```
Comment (Mahmoudzaid2994) [inline src/components/AgentCardsContainer/index.jsx:34]:
```suggestion
          <Wrapper>
            <AgentCard
              address={agent.address}
              agentImg={agent.agentImg}
              agentName={agent.agentName}
              companyName={agent.companyName}
              email={agent.email}
              key={agent.id}
              phoneNumber={agent.phoneNumber}
            />
          </Wrapper>
```

File: src/components/AgentsListBanner/index.jsx [hunk 24]
```diff
@@ -0,0 +1,87 @@
+import { useMediaQuery } from '@material-ui/core';
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:1]: Remove `material-ui` and handle the display for desktop or mobile using css

File: src/components/AgentsListBanner/index.jsx [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+                <option value="agent">By Agent Name</option>
+              </SelectFilter>
+            </FilterProperty>
+            {isDesktop ? (
+              <>
+                <SelectType>
+                  <option value="">Select Type</option>
+                </SelectType>
+                <LocationSearch placeholder="Search for Location" />
+                <AgentSearch placeholder="Agent's Name" />
```
Comment (Mahmoudzaid2994) [inline src/components/AgentsListBanner/index.jsx:60]: use shared input

File: src/styles/breakpoints.js [hunk 26]
```diff
@@ -22,6 +22,9 @@ const breakpoints = {
   // 1200px
   desktopMax: '75rem',
 
+  // 1440px
+  desktopExtraMax: '90rem',
+
```
Comment (Mahmoudzaid2994) [inline src/styles/breakpoints.js:27]:
```suggestion

```

File: src/components/agent-list/AgentCard/index.jsx [hunk 27]
```diff
@@ -0,0 +1,154 @@
...
+ *
+ * @return {JSX.Element}             The agent card holding all of the agent's information.
+ */
+function AgentCard({
+  address,
+  agentImg,
+  agentName,
+  companyName,
+  email,
+  phoneNumber,
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/index.jsx:54]:
```suggestion
data
```

File: src/components/agent-list/AgentCard/index.jsx [hunk 28]
```diff
@@ -0,0 +1,154 @@
...
+              onClick={() => {}}
+              padding="0 0.375rem"
+            >
+              <LikeIcon />
+            </RoundedButton>
+            <RoundedButton
+              ariaLabel="Share"
+              backgroundColor={colors.white}
+              height="1.875rem"
+              onClick={() => {}}
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/index.jsx:90]:
```suggestion
```

File: src/components/agent-list/AgentCard/index.jsx [hunk 29]
```diff
@@ -0,0 +1,154 @@
...
+  agentName,
+  companyName,
+  email,
+  phoneNumber,
+}) {
+  return (
+    <Card>
+      <ImgContainer>
+        <AgentPicture src={agentImg} />
+        <HeartButton
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/index.jsx:60]:
```suggestion
        <RoundedButton
```

File: src/components/agent-list/AgentCard/agent-card.styles.js [hunk 30]
```diff
@@ -0,0 +1,186 @@
...
+  color: ${colors.dustyGrey};
+  margin: 0.125rem;
+
+  :hover {
+    background-color: ${colors.green};
+    color: ${colors.white};
+  }
+`;
+
+export const HeartButton = styled(Button)`
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/agent-card.styles.js:162]:
```suggestion
export const HeartButton = styled(RoundedButton)`
```

File: src/components/agent-list/AgentCard/agent-card.styles.js [hunk 31]
```diff
@@ -0,0 +1,186 @@
...
+  text-decoration: underline;
+`;
+
+export const Line = styled.hr`
+  border: 0.0625rem solid ${colors.mercury};
+  margin: 0;
+`;
+
+export const HeartIcon = styled(heart)`
+  width: 1.125rem;
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/agent-card.styles.js:138]:
```suggestion
  width: 1rem;
```

File: src/components/AgentsListBanner/index.jsx (hunk 25 above)
Comment (RaniaSAloun) [reply src/components/AgentsListBanner/index.jsx]: the shared input is not suitable for the filter component

File: src/components/agent-list/AgentCard/agent-card.styles.js [hunk 32]
```diff
@@ -0,0 +1,186 @@
...
+  align-items: center;
+  display: flex;
+  height: 3.89rem;
+  justify-content: space-between;
+  margin: 0 1rem;
+`;
+
+export const IconContainer = styled.div`
+  width: 1.375rem;
+`;
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/agent-card.styles.js:186]:
```suggestion
export const IconContainer = styled.div`
  display: flex;

  > svg {
    width: 1rem;
    margin: 0.5rem;
  }
`;
```

File: src/components/agent-list/AgentCard/index.jsx [hunk 33]
```diff
@@ -0,0 +1,154 @@
...
+          <Button height="2rem" onClick={() => {}}>
+            Contact
+          </Button>
+          <FooterButtonsContainer>
+            <IconContainer>
+              <MailIcon />
+            </IconContainer>
+            <IconContainer>
+              <PhoneIcon />
+            </IconContainer>
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/index.jsx:137]:
```suggestion
            <IconContainer>
              <MailIcon />
              <PhoneIcon />
            </IconContainer>
```

File: src/components/agent-list/AgentsListBanner/agents-list-banner.styles.js [hunk 34]
```diff
@@ -0,0 +1,172 @@
...
+  `}
+
+  ${mq.tablet`
+    padding: 0.625rem;
+    text-align: center; 
+    width: 100%;
+  `};
+`;
+
+export const AgentSearch = styled.input`
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentsListBanner/agents-list-banner.styles.js:128]: Use only one component please
```suggestion
export const SearchTag = styled.input`
```

File: src/pages/AgentList.jsx [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Agent's List page component.
+ *
+ * @return {JSX.Element}
+ */
+function AgentList() {
+  return (
+    <>
+      <AgentsListBanner />
+      <AgentCardsContainer />
```
Comment (AhmadAlShwaiki) [inline src/pages/AgentList.jsx:14]:
```suggestion
  const [agents, setAgents] = useState([]);

  /**
   * Fetches agents data from API and sets it to agents state.
   */
  useEffect(() => {
    const fetchAgents = async () => {
      const data = await getAgents();
      setAgents(data);
    };
    fetchAgents();
  });

  return (
    <Container>
      <CardsContainer>
        {agents.map((agent) => (
          <Wrapper key={agent.id}>
            <AgentCard
              address={agent.address}
              agentImg={agent.agentImg}
              agentName={agent.agentName}
              companyName={agent.companyName}
              email={agent.email}
              phoneNumber={agent.phoneNumber}
            />
          </Wrapper>
        ))}
      </CardsContainer>
    </Container>

```

**Review by AhmadAlShwaiki — CHANGES_REQUESTED** [review]
components grouping and css cleanup

File: src/components/agent-list/AgentCard/agent-card.styles.js [hunk 36]
```diff
@@ -0,0 +1,182 @@
...
+
+export const ButtonsContainer = styled.div`
+  align-items: flex-start;
+  display: flex;
+  justify-content: space-between;
+  width: 3.75rem;
+`;
+
+export const FooterButtonsContainer = styled.div`
```
Comment (AhmadAlShwaiki) [inline src/components/agent-list/AgentCard/agent-card.styles.js:79]:
```suggestion
export const FooterButtonsContainer = styled(ButtonsContainer)`
  align-items: center;
```

**Review by AhmadAlShwaiki — APPROVED** [review]
small change requested, looks good.

**Review by AseelRousan-op — APPROVED** [review]

**Review by mohdabed123 — APPROVED** [review]
LGTM

**Review by AhmadAlShwaiki — APPROVED** [review]

**Pull request merged**